src/redis_cmd.cc

```diff
@@ -73,29 +74,47 @@ const char *errUnbalacedStreamList =
 const char *errTimeoutIsNegative = "timeout is negative";
 const char *errLimitOptionNotAllowed = "syntax error, LIMIT cannot be used without the special ~ option";
 
+enum class AuthResult {
+  OK,
+  INVALID_PASSWORD,
+  NO_REQUIRE_PASS,
+};
+
+AuthResult AuthenticateUser(Connection *conn, Config* config, const std::string& user_password) {
+  auto iter = config->tokens.find(user_password);
+  if (iter != config->tokens.end()) {
+    conn->SetNamespace(iter->second);
+    conn->BecomeUser();
+    return AuthResult::OK;
+  }
+  const auto& requirepass = config->requirepass;
+  if (!requirepass.empty() && user_password != requirepass) {
+    return AuthResult::INVALID_PASSWORD;
+  }
+  conn->SetNamespace(kDefaultNamespace);
+  conn->BecomeAdmin();
+  if (requirepass.empty()) {
+    return AuthResult::NO_REQUIRE_PASS;
+  }
+  return AuthResult::OK;
+}
+
```

Comment:
Hi. May I ask why do we introduce global `enum class AuthResult` and `AuthenticateUser`  function instead of putting them in classes or corresponding constant definition files? Thank you for your contribution.

Reply:
It's better to put `AuthenticateUser` and status to `Connection`, but here I just want to make it lightweight to implement it, because `auth` isn't widely used.

Reply:
I think we can at least put them in `class CommandAuth`.

Reply:
```
class CommandAuth : public Commandar {
 public:
   static void AuthenticateUser() ...
};

class CommandHello: public Commandar {
 public:
    Status Exec() {
        CommandAuth::..
    }
};

```

The code like that would be confusing, because `auth` is not a part of `Command`. And put it into client needs `Config` in server. So here I just implement it as helper function

Reply:
I noticed that the helper function of the existed class in `redis_cmd.cc` is also written in the class. Hi @git-hulk @PragmaTwice, What do you think？

Reply:
I think it is OK to me. Actually I think it can be `Status AuthenticateUser(Connection *conn, Config* config, const std::string& user_password)`, but current form also looks good to me.

Reply:
Both of them are ok to me